src/discussions/data/hooks.test.jsx

```diff
@@ -82,4 +83,101 @@ describe('Hooks', () => {
+      return (
+        <div>
+          {String(userCanAddThreadInBlackoutDate)}
+        </div>
+      );
+    }
+
+    function renderComponent() {
+      return render(
+        <IntlProvider locale="en">
+          <AppProvider store={store}>
+            <ComponentWithHook />
+          </AppProvider>
+        </IntlProvider>,
+      );
+    }
+    describe('blackout dates are not active', () => {
+      beforeEach(async () => {
+        store = initializeStore({
+          config: {
+            blackouts: [],
+            hasModerationPrivileges: false,
+            isGroupTa: false,
+            isCourseAdmin: false,
+            isCourseStaff: false,
+            isUserAdmin: false,
+          },
+        });
+      });
+      test('return true when blackout dates are not active and Role is Learner', () => {
+        const { queryByText } = renderComponent();
+        expect(queryByText('true')).toBeInTheDocument();
+      });
+    });
+
+    describe('blackout dates are active', () => {
+      beforeEach(async () => {
+        store = initializeStore({
+          config: {
+            blackouts: [{ start: '2022-11-25T00:00:00Z', end: '2050-11-25T23:59:00Z' }],
+            hasModerationPrivileges: false,
+            isGroupTa: false,
+            isCourseAdmin: false,
+            isCourseStaff: false,
+            isUserAdmin: false,
+          },
+        });
+      });
+
+      test('return false when blackout dates are active and Role is Learner', async () => {
+        const { queryByText } = renderComponent();
+        expect(queryByText('false')).toBeInTheDocument();
+      });
+    });
+    describe('blackout dates are active And role is other than Leaner', () => {
+      beforeEach(async () => {
+        store = initializeStore({
+          config: {
+            blackouts: [{ start: '2022-11-25T00:00:00Z', end: '2050-11-25T23:59:00Z' }],
+            hasModerationPrivileges: false,
+            isGroupTa: false,
+            isCourseAdmin: true,
+            isCourseStaff: false,
+            isUserAdmin: false,
+          },
+        });
+      });
+
+      test('return true when blackout dates are active and Role is not Learner', async () => {
+        const { queryByText } = renderComponent();
+        expect(queryByText('true')).toBeInTheDocument();
+      });
+    });
+    describe('blackout dates are not active And role is other than Leaner', () => {
+      beforeEach(async () => {
+        store = initializeStore({
+          config: {
+            blackouts: [],
+            hasModerationPrivileges: false,
+            isGroupTa: false,
+            isCourseAdmin: true,
+            isCourseStaff: false,
+            isUserAdmin: false,
+          },
+        });
+      });
+
+      test('return true when blackout dates are not active and Role is not Learner', async () => {
+        const { queryByText } = renderComponent();
+        expect(queryByText('true')).toBeInTheDocument();
+      });
+    });
+  });
```

Comment:
1. remove beforeEach from each test function as it contains only 1 test case.
2. remove async keyword from beforeEach function
3. structural changes

` describe('blackout dates are active', () => {
      beforeEach(async () => {
        store = initializeStore({
          config: {
            blackouts: [{ start: '2022-11-25T00:00:00Z', end: '2050-11-25T23:59:00Z' }],
            hasModerationPrivileges: false,
            isGroupTa: false,
            isCourseAdmin: false,
            isCourseStaff: false,
            isUserAdmin: false,
          },
        });
      });

      test('return false when blackout dates are active and Role is Learner', async () => {
        const { queryByText } = renderComponent();
        expect(queryByText('false')).toBeInTheDocument();
      });
       test('return false when blackout dates are active and Role is other than', async () => {
        const { queryByText } = renderComponent();
        expect(queryByText('false')).toBeInTheDocument();
      });
    });`


` describe('blackout dates are inActive', () => {
      beforeEach(async () => {
        store = initializeStore({
          config: {
            blackouts: [{ start: '2022-11-25T00:00:00Z', end: '2050-11-25T23:59:00Z' }],
            hasModerationPrivileges: false,
            isGroupTa: false,
            isCourseAdmin: false,
            isCourseStaff: false,
            isUserAdmin: false,
          },
        });
      });

      test('return {{true}} when blackout dates are active and Role is Learner', async () => {
       
      });
       test('return {{true}} when blackout dates are active and Role is not Learner', async () => {
       
      });
    });`